File: ios/CortexV2-Emotiv-iOS/CortexV2-Emotiv-ObjC/CortexV2-Emotiv-ObjC/Headset.m
//
//  Headset.m
//  CortexV2-Emotiv-ObjC
//
//  Created by nvtu on 2/24/20.
//  Copyright © 2020 Emotiv. All rights reserved.
//

#import "Headset.h"

@implementation Headset

- (id) init {
    self = [super init];
    if (self) {
        headsetId = @"";
        label = @"";
        connectedBy = @"";
        status = @"";
    }
    return self;
}

- (id) initJson : (NSDictionary *) jHeadset {
    self = [super init];
    if (self) {
        headsetId = [jHeadset objectForKey:@"id"];
        label = [jHeadset objectForKey:@"label"];
        connectedBy = [jHeadset objectForKey:@"connectedBy"];
        status = [jHeadset objectForKey:@"status"];
    }
    return self;
}

-(NSString *) toString {
    return [NSString stringWithFormat:@"%@ %@ %@", headsetId, status, connectedBy];
}
@end

File: mobile/ios/CortexV2-Emotiv-iOS/CortexV2-Example-Swift/CortexV2-Example-Swift/Headset.swift
//
//  Headset.swift
//  CortexV2-Example-Swift
//
//  Created by Emotiv Inc on 2/21/20.
//  Copyright © 2020 Emotiv. All rights reserved.
//

import Foundation

class Headset {
    var id: String = ""
    var label: String = ""
    var connectedBy: String = ""
    var status: String = ""
    
    init() {
        
    }
    
    init(jHeadset: [String: Any]) {
        fromJson(jheadset: jHeadset)
    }
    
    func fromJson(jheadset: [String: Any]) {
        id = jheadset["id"] as? String ?? ""
        label = jheadset["label"] as? String ?? ""
        connectedBy = jheadset["connectedBy"] as? String ?? ""
        status = jheadset["status"] as? String ?? ""
    }
    
    func toString() -> String {
        return "\(id) \(status) \(connectedBy)"
    }
}
